src/snowflake/snowpark/session.py

```diff
@@ -851,17 +852,16 @@ def _resolve_packages(
                 package_name = MODULE_NAME_TO_PACKAGE_NAME_MAP.get(
                     package.__name__, package.__name__
                 )
-                package = f"{package_name}=={pkg_resources.get_distribution(package_name).version}"
+                package = f"{package_name}=={importlib.metadata.distribution(package_name).version}"
                 use_local_version = True
             else:
                 package = package.strip().lower()
                 use_local_version = False
-            package_req = pkg_resources.Requirement.parse(package)
-            # get the standard package name if there is no underscore
-            # underscores are discouraged in package names, but are still used in Anaconda channel
-            # pkg_resources.Requirement.parse will convert all underscores to dashes
+            package_req = Requirement(package)
             package_name = (
-                package if not use_local_version and "_" in package else package_req.key
+                package
+                if not use_local_version and "_" in package
+                else package_req.name.lower().replace("_", "-")
```

Comment:
Sanity check: Is `package_req.name.lower().replace("_", "-")` the same as `package_req.key`?

Reply:
According to https://setuptools.pypa.io/en/latest/pkg_resources.html#requirement-methods-and-attributes, `key` is "an all-lowercase version of the project_name, useful for comparison or indexing". The inline comment suggests it also replaces `_` with `-`.